packages/devtools_app/lib/src/screens/memory/framework/memory_controller.dart

```diff
@@ -85,55 +85,56 @@ class MemoryController extends DisposableController
     @visibleForTesting DiffPaneController? connectedDiff,
     @visibleForTesting ProfilePaneController? connectedProfile,
   ) async {
-    assert(!_initialized.isCompleted);
+    assert(!_dataInitialized.isCompleted);
     switch (_mode) {
       case DevToolsMode.disconnected:
         // TODO(polina-c): load memory screen in disconnected mode, https://github.com/flutter/devtools/issues/6972
-        _initializeData();
+        await _initializeData();
       case DevToolsMode.connected:
-        _initializeData(
+        await _initializeData(
           diffPaneController: connectedDiff,
           profilePaneController: connectedProfile,
         );
       case DevToolsMode.offlineData:
         assert(connectedDiff == null && connectedProfile == null);
-        await maybeLoadOfflineData(
+        final loaded = await maybeLoadOfflineData(
           ScreenMetaData.memory.id,
           createData: (json) {
             final data = json[_jsonKey];
             if (data is OfflineMemoryData) return data;
             return OfflineMemoryData.fromJson(data as Map<String, dynamic>);
           },
           shouldLoad: (data) => true,
-          loadData: (data) => _initializeData(offlineData: data),
+          loadData: (data) async => await _initializeData(offlineData: data),
         );
         // [maybeLoadOfflineData] will be a noop if there is no offline data for the memory screen,
         //  so ensure we still call [_initializedData] if it has not been called.
-        if (!_initialized.isCompleted) _initializeData();
-        assert(_initialized.isCompleted);
+        assert(loaded == _dataInitialized.isCompleted);
+        if (!loaded) {
```

Comment:
Now that we have the assert, we want to be checking `_dataInitialized.isCompleted` here instead.